src/net/sourceforge/kolmafia/request/AdventureRequest.java

```diff
@@ -669,9 +669,18 @@ public static String parseChoiceEncounter(
     }
 
     if (urlString.equals("choice.php")) {
-      // Since we cannot walk from this choice, this must have been KoL
-      // redirecting back to the choice after a page refresh
-      return null;
+      // Although KoL usually redirects to choice.php?forceoption=0, older
```

Comment:
this is unfortunate :(

I think the only solution would be to note down where it happens and do some special exceptions, or try to get CDM involved?

Reply:
I think I figured out a workaround

 If you get exactly "choice.php" and your last encounter is the same as this one, it is a refresh.

I don't think there will be:

choice.php (encounter name "Hello")
choice.php?whichchoice=X&option=Y
choice.php (encounter name "Hello")

I'll write some tests for this and adjust it.

Reply:
Yes! When in Mole or Bird form, you adventure for 12 or 15 turns and the next adventure you take redirects to "choice.php" to the "Welcome Back!" encounter.

I used the HTML for leaving reincarnation as a mole and verified that "Welcome Back!" got added as an encounter.
(In the relay browser I refreshed that page multiple times and it did not count the refreshes as new encounters)
And I restored the check in the ChoiceManagerTest for detecting that situation.